server.js

```diff
@@ -11,13 +16,165 @@ const welcomeMessage = {
   text: "Welcome to CYF chat system!",
 };
 
-//This array is our "data store".
-//We will start with one message in the array.
-//Note: messages will be lost when Glitch restarts our server.
 const messages = [welcomeMessage];
 
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+app.get("/messages", function (req, res) {
+  res.send(messages);
+});
+
+app.get("/messages/:id", (req, res) => {
+  const getById = Number(req.params.id);
+  const findByID = messages.find((message) => message.id === getById);
+  if (!findByID)
+    return res.status(404).send("Message with the given ID was not found");
+  res.send(findByID);
+});
+
+app.post("/messages", (req, res) => {
+  const { from, text } = req.body;
+
+  // Check if the 'from' and 'text' properties are present and not empty
+  if (!from || !text || from.trim() === "" || text.trim() === "") {
```

Comment:
This is very smart. Just because we don't have power over what user may type, we are anticipating and trimming any excess whitespace. Great work 💪